packages/spanner/src/transaction.js

```diff
@@ -223,25 +228,26 @@ Transaction.prototype.begin = function(callback) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
```

Comment:
I think those are fair points, I am hesitant because I already see this being an annoyance for Promise users. I'm guessing if we were to just not support Promises here, we would see a bunch of users doing something like this..

```js
function runTransaction() {
  return new Promise((resolve, reject) => {
    database.runTransaction((err, transaction) => {
      if (err) return reject(err);

      // do a bunch of transactional chaining..
      transaction.commit().then(resolve, reject);
    });
  });
}
```

This makes me sad. It might be because we make the user manually call `commit`, but I feel like we should be able to offer Promise users something similar to this - I'm just not sure what the ideal way to go about this is.